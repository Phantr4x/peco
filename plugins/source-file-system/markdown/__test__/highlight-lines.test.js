const markdown = require('markdown-it')
const plugin = require('../highlight-lines')

const text = `
\`\`\`html {2}
<template>
  <div>{{ msg }}<div>
</template>

<script>
export default {
  data() {
    return {
      msg: 'hello'
    }
  }
}
</script>
\`\`\`

\`\`\`js
<script>
hehe
</script>
\`\`\`
`

test('main', () => {
  const md = markdown({
    highlight: require('../highlight')
  })
  md.use(plugin)
  const html = md.render(text)
  expect(html).toMatchSnapshot()
})
