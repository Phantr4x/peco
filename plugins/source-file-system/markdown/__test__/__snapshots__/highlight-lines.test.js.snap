// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
"<div class=\\"pre-wrapper\\" v-pre=\\"\\" data-lang=\\"html\\"><div class=\\"code-mask\\"><span class=\\"code-line\\">&lt;template&gt;</span><span class=\\"code-line highlighted\\">  &lt;div&gt;{{ msg }}&lt;div&gt;</span><span class=\\"code-line\\">&lt;/template&gt;</span><span class=\\"code-line\\">&#8203;</span><span class=\\"code-line\\">&lt;script&gt;</span><span class=\\"code-line\\">export default {</span><span class=\\"code-line\\">  data() {</span><span class=\\"code-line\\">    return {</span><span class=\\"code-line\\">      msg: 'hello'</span><span class=\\"code-line\\">    }</span><span class=\\"code-line\\">  }</span><span class=\\"code-line\\">}</span><span class=\\"code-line\\">&lt;/script&gt;</span><span class=\\"code-line\\">&#8203;</span></div><pre class=\\"language-html\\"><code class=\\"language-html\\"><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>template</span><span class=\\"token punctuation\\">></span></span>
  <span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>div</span><span class=\\"token punctuation\\">></span></span>{{ msg }}<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>div</span><span class=\\"token punctuation\\">></span></span>
<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>template</span><span class=\\"token punctuation\\">></span></span>

<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>script</span><span class=\\"token punctuation\\">></span></span><span class=\\"token script language-javascript\\">
<span class=\\"token keyword\\">export</span> <span class=\\"token keyword\\">default</span> <span class=\\"token punctuation\\">{</span>
  <span class=\\"token function\\">data</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>
    <span class=\\"token keyword\\">return</span> <span class=\\"token punctuation\\">{</span>
      msg<span class=\\"token punctuation\\">:</span> <span class=\\"token string\\">'hello'</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>
<span class=\\"token punctuation\\">}</span>
</span><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>script</span><span class=\\"token punctuation\\">></span></span></code></pre></div><div class=\\"pre-wrapper\\" v-pre=\\"\\" data-lang=\\"js\\"><pre class=\\"language-js\\"><code class=\\"language-js\\"><span class=\\"token operator\\">&lt;</span>script<span class=\\"token operator\\">></span>
hehe
<span class=\\"token operator\\">&lt;</span><span class=\\"token operator\\">/</span>script<span class=\\"token operator\\">></span></code></pre></div>"
`;
